mlflow/sagemaker/container/__init__.py

```diff
@@ -97,6 +104,96 @@ def _serve():
+        print_flush("The version of python used to serialize the model, Python {mpyv}, is not"
+                     " supported by Anaconda.".format(mpyv=model_py_version))
+        _warn_potentially_incompatible_conda_env()
+        return None
+    elif model_py_version == PYTHON_VERSION:
+        print_flush("The model's Python version matches the default environment's"
+                    " Python version: {dpyv}. Using the default environment.".format(
+                        dpyv=DEFAULT_CONDA_PYTHON_VERSION))
+        return None
+    else:
+        print_flush("The model's Python version is {mpyv}. Activating Anaconda environment with"
+                    " Python version {mpyv}.".format(mpyv=model_py_version))
+        return CondaEnv(name="py_env", config=None, py_version=model_py_version)
+
+
+def _add_py_version_to_conda_env_if_necessary(env_path, model_py_version):
+    try:
+        with open(env_path, "r") as f:
+            env_yaml = yaml.load(f)
+
+        deps = env_yaml["dependencies"]
+        env_py_versions = list(filter(
+            lambda item : "=" in item and item.split("=")[0].lower() == "python", deps))
+        env_contains_python = len(env_py_versions) > 0
+        if env_contains_python:
+            env_py_version = env_py_versions[-1].split("=")[1]
+            print_flush("Using Python version specified by the custom Anaconda environment:" 
+                        " Python {epyv}.".format(epyv=env_py_version))
+        else:
+            print_flush("The custom Anaconda environment does not specify a version of Python to use.")
+            if model_py_version is not None:
+                print_flush("Using the version of the Python associated with the model:" 
+                            " Python {mpyv}".format(mpyv=model_py_version))
+                deps.append("python={mpyv}".format(mpyv=model_py_version))
+            else:
+                deps.append("python={dpyv}".format(dpyv=DEFAULT_CONDA_PYTHON_VERSION))
+                _warn_potentially_incompatible_conda_env()
+
+        with open(env_path, "w") as out:
+            yaml.dump(env_yaml, out, default_flow_style=False)
+    except Exception as e:
+        print(e)
+
+
+def _create_conda_env(env):
+    if env.config is not None:
+        cmd = "conda env create -n {en} -f {cfg}".format(en=env.name, cfg=env.config)
+    elif env.py_version is not None:
+        cmd = "conda create -n {en} python={pyv}".format(en=env.name, pyv=env.py_version)
+    os.system(cmd)
+
+
+def _conda_supports_py_version(py_version):
+    major_version = ".".join(py_version.split(".")[:2])
+    return (major_version in SUPPORTED_CONDA_MAJOR_PY_VERSIONS)
+
+
+def _warn_potentially_incompatible_conda_env():
+    """
+    Prints a warning message indicating that the default Anaconda environment,
+    which may not be compatible with the model, will be used for serving.
+    """
+    print_flush("Using the default Anaconda environment with Python {dpyv}, which may not be"
```

Comment:
Maybe `[WARNING]` or something? I think there's a lot of logging messages expected by default, so might be good to point this one out.